dali/pipeline/workspace/device_workspace.cc

```diff
@@ -36,6 +36,24 @@ const TensorList<GPUBackend>& DeviceWorkspace::Input(int idx) const {
   return *gpu_inputs_[tensor_meta.second];
 }
 
+template <>
+TensorList<CPUBackend>& DeviceWorkspace::MutableInput(int idx) {
```

Comment:
Can't we extract the common part from  DeviceWorkspace::Input and DeviceWorkspace::MutableInput as a helper?

Reply:
done